dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/log/remote/GcsRemoteLogHandler.java

```diff
@@ -0,0 +1,153 @@
+    }
+
+    public void init() {
+        try {
+            credential = readCredentials();
+            bucketName = readBucketName();
+            gcsStorage = buildGcsStorage(credential);
+
+            checkBucketNameExists(bucketName);
+        } catch (IOException e) {
+            log.error("GCS Remote Log Handler init failed", e);
+        }
+    }
+
+    @Override
+    public void close() throws IOException {
+        try {
+            if (gcsStorage != null) {
+                gcsStorage.close();
+            }
+        } catch (Exception e) {
+            throw new IOException(e);
+        }
+    }
+
+    @Override
+    public void sendRemoteLog(String logPath) {
+        String objectName = RemoteLogUtils.getObjectNameFromLogPath(logPath);
+
+        try {
+            log.info("send remote log {} to GCS {}", logPath, objectName);
+
+            BlobInfo blobInfo = BlobInfo.newBuilder(
+                    BlobId.of(bucketName, objectName)).build();
+
+            gcsStorage.create(blobInfo, Files.readAllBytes(Paths.get(logPath)));
+        } catch (Exception e) {
+            log.error("error while sending remote log {} to GCS {}", logPath, objectName, e);
+        }
+    }
+
+    @Override
+    public void getRemoteLog(String logPath) {
+        String objectName = RemoteLogUtils.getObjectNameFromLogPath(logPath);
+
+        try {
+            log.info("get remote log on GCS {} to {}", objectName, logPath);
+
+            Blob blob = gcsStorage.get(BlobId.of(bucketName, objectName));
+            blob.downloadTo(Paths.get(logPath));
+        } catch (Exception e) {
+            log.error("error while getting remote log on GCS {} to {}", objectName, logPath, e);
+        }
+    }
+
+    protected Storage buildGcsStorage(String credential) throws IOException {
+        return StorageOptions.newBuilder()
+                .setCredentials(ServiceAccountCredentials.fromStream(
+                        Files.newInputStream(Paths.get(credential))))
+                .build()
+                .getService();
+    }
+
+    protected String readCredentials() {
+        return PropertyUtils.getString(Constants.REMOTE_LOGGING_GCS_CREDENTIAL);
+    }
+
+    protected String readBucketName() {
+        return PropertyUtils.getString(Constants.REMOTE_LOGGING_GCS_BUCKET_NAME);
+    }
+
+    public void checkBucketNameExists(String bucketName) {
+        if (StringUtils.isBlank(bucketName)) {
+            throw new IllegalArgumentException(Constants.REMOTE_LOGGING_GCS_BUCKET_NAME + " is blank");
+        }
+
+        boolean exist = false;
+        for (Bucket bucket : gcsStorage.list().iterateAll()) {
+            if (bucketName.equals(bucket.getName())) {
+                exist = true;
+                break;
+            }
+        }
+
+        if (!exist) {
+            throw new IllegalArgumentException(
+                    "bucketName: " + bucketName + " is not exists, you need to create them by yourself");
+        } else {
```

Comment:
I think it does't need to throw exception here if not exist. WDYT?

Reply:
Thanks for your comment. The exception thrown here is the same as in the resource center (also in the `checkBucketNameExists()` OSS / GCS / S3).
I am not sure if we need to delete them all.